cmd/engine/engine_test.go

```diff
@@ -88,3 +91,50 @@ func Test_doCmd(t *testing.T) {
 		})
 	}
 }
+
+func Test_doCMDWithDiscovery(t *testing.T) {
```

Comment:
Personally, I am not sure if there is much sense in testing this in "main". In the long run it would probably be better to extract functionality out of our main method into smaller functions and make them testable, but going through hoops to have some testing function just for the sake of having tests in main is counter-productive in my opinion

Reply:
Can I somehow hard-reset to 407c26af4be03503647d8020e0336546ba0f77de (Update cmd/engine/engine.go)?